app/dojo.py

```diff
@@ -199,3 +198,62 @@ def print_unallocated(self, filename):
 			txt_file.write(output)
 			txt_file.close()
 			return("Data saved in {}.txt \n".format(filename) )
+
+	def reallocate_person(self, personID, roomname):
+		""" Reallocatesa person to a different room """
+		#returns rooms with occupants
+		room_with_occupants = [room for room in self.all_rooms\
+		if len(room.occupants) > 0]
+
+		person_reallocate = [person for person \
+		in self.all_people if person.id == personID]
+		#checks if person is in allocated a room
+		previous_room = [room for room in room_with_occupants\
+		 for person in room.occupants if person.id == personID]
+		#checks if room to rellocate to exists
+		new_room = [room for room in self.all_rooms \
+		if room.room_name.lower() == roomname.lower()]
+		if new_room:
+			#checks if previous room type matches with new room type
+			roomtypes = [oldroom for oldroom in \
+			previous_room if oldroom.room_type == new_room[0].room_type]
+
+		if not person_reallocate:
```

Comment:
maybe the return message indicating person was not found would be preferred, since the ID could be of valid type but it doesn't exist. Then have a different statement to check if personID is valid or not